test/join.jl

```diff
@@ -216,20 +216,19 @@ module TestJoin
 
     @testset "all joins with CategoricalArrays" begin
         dt1 = DataTable(Any[CategoricalArray([1, 3, 5]),
-                              CategoricalArray([1.0, 3.0, 5.0])], [:id, :fid])
+                            CategoricalArray([1.0, 3.0, 5.0])], [:id, :fid])
         dt2 = DataTable(Any[CategoricalArray([0, 1, 2, 3, 4]),
-                              CategoricalArray([0.0, 1.0, 2.0, 3.0, 4.0])], [:id, :fid])
+                            CategoricalArray([0.0, 1.0, 2.0, 3.0, 4.0])], [:id, :fid])
         N = null
-        DRT = CategoricalArrays.DefaultRefType
 
         @test join(dt1, dt2, kind=:cross) ==
             DataTable(Any[repeat([1, 3, 5], inner = 5),
                           repeat([1, 3, 5], inner = 5),
                           repeat([0, 1, 2, 3, 4], outer = 3),
                           repeat([0, 1, 2, 3, 4], outer = 3)],
                       [:id, :fid, :id_1, :fid_1])
-        @test typeof.(join(dt1, dt2, kind=:cross).columns) ==
-            [CategoricalVector{i, DRT} for i in [Int, Float64, Int, Float64]]
+        @test all(map(<:, typeof.(join(dt1, dt2, kind=:cross).columns),
```

Comment:
This allows us to avoid checking for equality with the DefaultRefType in CategoricalArrays along with the 3rd and 4th parameter type, but strangely just using `all` results in an error. It requires `map` then `all`. Not sure what the bug is.